lib/server.js

```diff
@@ -1435,7 +1435,7 @@ Server.prototype._routeErrorResponse = function _routeErrorResponse(
         }
 
         // only automatically send errors that are known (e.g., restify-errors)
-        if (err instanceof Error && _.isNumber(err.statusCode)) {
+        if (err instanceof Error && _.isNumber(err.statusCode || err.status)) {
```

Comment:
when is this `status`? I 
`restify-errors` does `statusCode`
https://github.com/restify/errors/blob/master/lib/baseClasses/HttpError.js#L54

Reply:
exactly, and `restify-errors` doesn't conform to rfc7807, which defines `status`. The problem is that it assumes that users are indeed using `restify-errors` and don't give them the ability to follow the standard.

This PR keeps `statusCode` for backward compatibility with `restify-errors` but allows restify to return the correct content type `application/problem+json` when `status` is used and no `statusCode` is present

Reply:
```js
class NotFound extends Error {
  constructor(detail) {
    this.title = 'Not Found';
    this.type = 'https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404';
    this.status = 404;
    this.detail = detail || 'Resource not found';
  }
  
  toJSON() {
    return this;
  }
}
```

this will conform with rfc7807 and will hit that code with `status`

Reply:
You know what? It doesn't really need to "have a status property instead of a statusCode" to conform to the rfc. I was mainly using `status` property as a way to identify that was an error message complying with the rfc and sending the correct content-type.

But we can have a `statusCode` property in the error object and on the `toJSON` method that I serialize it to `status` instead, but I would need to find a way to set the `application/problem+json` content type and still keep backward compatibility. Any thoughts on this?